### DISCUSSION TITLE
TS type inference when selecting foreign tables (Many-to-1)

### DISCUSSION THREAD

**Discussion by Rizzaxc** [supabase/supabase#11299] [Questions] [answered]
I'm taking an example from the POSTgrest doc: https://postgrest.org/en/stable/api.html#resource-embedding

When querying for `director` from `films`, I can't seem to get correct TS typing from the generated types. The typing for `data.director` says 
`director: {id, name} | director: {id, name}[] | null `
while I'd prefer it to only contain the singular variant, since my database schema enforces `NOT NULL` and the relationship is many to one (no plurals).

This guide https://supabase.com/docs/reference/javascript/typescript-support doesn't help, unless I use type guard which is not very ergonomic 

-- reposting from Discord for traceability

**Comment by silentworks** [comment, marked as answer]
This is currently an issue with the generator. We will be working on fixing this in the future, but for now you can use a guard to narrow down the typings. An example of this would be
```ts
const { data, error } = await supabase
  .from("album")
  .select(`id, title, artist(id, name)`)
  .limit(1)
  .single();

if (data && data?.artist) { // here we guard to make sure its not null
  if (!Array.isArray(data.artist)) { // here we guard to make sure it isn't an array
    data.artist.name // now we should get correct typing here
  }
}
```
